src/coll/mod.rs

```diff
@@ -177,9 +177,9 @@ impl Collection {
         let aggregate = Aggregate::new(self.namespace(), pipeline, options);
         let client = self.client();
         client
-            .execute_operation(&aggregate, None)
+            .execute_operation_with_implicit_session(aggregate)
```

Comment:
A little update on this: `execute_cursor_operation` needs to return an `Option<ClientSession>` anyways since standalone topologies don't support sessions. So, now we're in a state where we could define a single executor entry point if we wanted to, something like:

```rust
pub(crate) fn execute_operation<T: Operation>(
    op: T, 
    options: Option<ExecutionOptions>
) -> Result<ExecutionResult<T::O>> { ... }

struct ExecutionOptions {
    session: Option<&mut ClientSession>,
    connection: Option<&mut Connection>
}

struct ExecutionResult<T> {
    result: T,
    implicit_session: Option<ClientSession>,
    selected_connection: Option<Connection>,
}
```
Though I think this work can be deferred to whenever we decide to modularize `execute_operation`.

Reply:
I'm fine with leaving it as-is until we decide to modularize `execute_operation` as you mentioned.